### ISSUE TITLE
Timer not in milliseconds

### ISSUE THREAD

**Issue by sapter** [bradgarropy/use-timer#1]
Hi Brad,

Thanks for writing up this nifty utility hook. I noticed that, contrary to the documentation in the readme, the timer is in centiseconds (1/100th of a second), rather than in milliseconds (1/1000th of a second). Can you please update the documentation to reflect that or set the interval in the code to update after 1 millisecond.

**Comment by bradgarropy** [comment]
Hey @sapter,
Great find! I think you're right, I totally messed that up. Let me try and get to fixing that tonight.

**Comment by bradgarropy** [comment]
I started looking into this a bit today and found that the browser limits the minimum interval to `4ms`. Check out the [Delay Restrictions](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval#delay_restrictions) section of the documentation.

That being said, when I set the interval to `1ms` the ticks looked like this.

```
elapsedTime: 0
elapsedTime: 1
elapsedTime: 2
elapsedTime: 3
elapsedTime: 4
elapsedTime: 8
elapsedTime: 12
elapsedTime: 16
elapsedTime: 20
```

Is that fine grain enough for you @sapter? Or are you looking for more?

**Comment by sapter** [comment]
The delay restrictions talk about _nested_ intervals (i.e. one `setInterval` calls another `setInterval`). Is that the case in our situation here? I don't see it but maybe I'm missing something?

Why `elapsedTime` starts incrementing by multiples of 4 after reaching `elapsedTime: 4` is a mystery to me... As far as I can see, there is nothing nested and the logic being executed is pretty simple and straightforward. It's also perfectly fine to have a timer that's accurate to +- 4 ms (or even to 100 ms as it's implemented now). Just reflect that in the readme/docs.

**Comment by sapter** [comment]
To clarify further, accuracy matters, In some applications, ±4ms is trivial (i.e running time in physical sports), but in others (such as measuring the performance of in-memory data access) it makes a big difference. You don't have to support all use cases, but be consistent and upfront about it.

**Comment by bradgarropy** [comment]
Yea, that behavior is pretty weird regarding the delay restrictions. I think I'll keep the `setInterval` at `10ms` increments and just document the precision in the [README](https://github.com/bradgarropy/use-timer/blob/master/readme.md).

I was also going to add a little section in there cautioning users to look elsewhere if they want something very precise. Then link over to the [Performance Timing](https://developer.mozilla.org/en-US/docs/Web/API/Performance) documentation.

What do you think about that approach?

**Comment by sapter** [comment]
Sounds good.

**Comment by bradgarropy** [comment]
Leaving this one open until I can make the relevant documentation changes.

**Comment by bradgarropy** [comment]
@all-contributors please add @sapter for bug and doc

**Comment by bradgarropy** [comment]
Closed by #2, released in [`v1.0.2`](https://github.com/bradgarropy/use-timer/releases/tag/v1.0.2).
